### ISSUE TITLE
Rendering problums

### ISSUE THREAD

**Issue by Gr33ntimer15** [HackerPoet/NonEuclidean#4]
So when i load the Engine it seems to have a rendering problem i don't know whats causing it My computers sepces are:

Win10(64 bit)
Amd A8-741 APU with AMD radeon R5
8GB of ram
(its a laptop)

i've also included an image of what it looks like

![1-nee](https://user-images.githubusercontent.com/46512055/51081263-80f7e000-16b9-11e9-93fc-d3c1382e9c74.PNG)

**Comment by Gr33ntimer15** [comment]
It was very hard to get that screen shot because the engine dosn't let you click off of it and there was no way of taking screen shots built in

**Comment by benelot** [comment]
This looks like a typical raw OpenGL problem. Especially if you use parts of the rendering engine in a special manner, you might get behavioral differences from device to device. At least that was my experience when we wrote a game at university from scratch. Maybe porting this into a game engine might help to avoid these differences.

**Comment by kolya5544** [comment]
Same for me. Very sad. Maybe needs self-compilation?

**Comment by Galaxtone** [comment]
Self-compilation?

**Comment by Galaxtone** [comment]
Bump

**Comment by Avieasolia** [comment]
This is interesting, I have the same problem and will look into it a bit more; I also have an AMD Radeon and this problem happened with a graphics engine I wrote in C# without even the use of OpenGL, I don't know if it was the same exact problem, but if anyone figures out a solution that will allow me to continue writing for OpenGL (I can't use other engines for what I'm working on) I would be very interested in any additional findings. Maybe something even mathematical or dealing with floating-point precision?

**Comment by ehx-v1** [comment]
I have a suggestion for narrowing this down. I'm not sure whether computers with extra GPU (like Nvidia or AMD) still have on-board GPU as well, I haven't investigated on hardware that much. If they do, you could just try to disable your AMD Radeon in the system settings (should be possible somehow) so it runs on the on-board GPU instead. It will probably lag horribly, but you should be able to make out whether the issue persists, and thereby whether or not it's actually GPU specific.

**Comment by Avieasolia** [comment]
Sorry it took so long, you were correct, it renders perfectly once the Radeon driver is turned off. This is a bit irritating since it's the better card for rendering in general, so I hope another fix comes around before I start getting lag because of the default card.

**Comment by Avieasolia** [comment]
Yeah they usually do have two gcards, you can enable or disable them through Device Manager -> Display Adapters

**Comment by ehx-v1** [comment]
So it's most likely some sort of issue in the AMD C# interface. I took a quick look on the Support Forums, but the only thing I discovered that sounds somewhat related is [this topic that sounds rather generic](https://community.amd.com/message/2907648). Not sure if anything has been discovered on that matter.

**Comment by ehx-v1** [comment]
Also, it might be possible to track down what kind of instructions or programming patterns cause the issue by trying different bits of code, but that sounds like a pain in the ass. One thing I'm fairly sure of, though, is that it's the AMD folks who messed up here.

**Comment by CrabAss** [comment]
I have also observed some rendering bugs occasionally when I tried to switch the scenes. Some "black surfaces" will be rendered randomly and unexpectedly and it always exists when I was moving my camera.

**Comment by AngryCarrot789** [comment]
i fixed this by updating my GPU drivers (more like installing the ones from the AMD website instead of the default windows drivers). i dont know why it does this but it's most likely the drivers
